### PR TITLE
partition a disk during install

### PR DESCRIPTION
**Pull request by hallyn**

### CODE REVIEW

**Review by smoser — COMMENTED** [review]
i think these comments are out of date...started typing yesterday. 

i had also started yesterday fixing some of the things myself.

File: cmd/mosctl/install.go [hunk 1]
```diff
@@ -18,8 +36,551 @@ var installCmd = cli.Command{
...
+	mysys := linux.System()
+	disks, err := getDiskSet(mysys, "all")
+	if err != nil {
+		return err
+	}
+
+	// We look for a disk which is > MinDiskSpace, and which either is
+	// empty or has just our SBF and PBF.
+	oDisks := []disko.Disk{}
+	for _, d := range disks {
```
[the remark this answers is not in the document]
Comment (hallyn) [reply cmd/mosctl/install.go:428]: Yes, it's doing both.  Are you asking me to fix the comment to reflect that?  Or to change the code to require a separate explicit wiping step?  If the latter, then at least we should search all the disks for existing partitions with our partlabels, and delete those.

File: cmd/mosctl/install.go [hunk 2]
```diff
@@ -10,6 +24,10 @@ var installCmd = cli.Command{
 	Usage:  "install a new mos system",
 	Action: doInstall,
 	Flags: []cli.Flag{
+		cli.BoolFlag{
+			Name: "partition",
+			Usage: "Disable to use existing partitioning",
+		},
 		cli.StringFlag{
 			Name:  "root, rfs, r",
```
[the remark this answers is not in the document]
Comment (hallyn) [reply cmd/mosctl/install.go:32]: The -r is really just for unprivileged testing of the manifest handling code.  Perhaps there's a better way to handle that.

File: cmd/mosctl/install.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (smoser) [reply cmd/mosctl/install.go:429]: addressed that https://github.com/project-machine/mos/pull/54

**Pull request merged**